src/Diagnostics/FxCop/Desktop.Analyzers/CSharp/MSInternal/DoNotCatchCorruptedStateExceptionsAnalyzer.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Collections.Immutable;
+
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+using SDL.Analyzer.Common;
+
+namespace Desktop.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class CSharpDoNotCatchCorruptedStateExceptionsAnalyzer : DoNotCatchCorruptedStateExceptionsAnalyzer
+    {
+        private static readonly ImmutableArray<SyntaxKind> nodeKindsOfInterest =
+            ImmutableArray.Create(SyntaxKind.MethodDeclaration,
+                                  SyntaxKind.ConstructorDeclaration,
+                                  SyntaxKind.DestructorDeclaration,
+                                  SyntaxKind.ConversionOperatorDeclaration,
+                                  SyntaxKind.OperatorDeclaration,
+                                  SyntaxKind.AddAccessorDeclaration,
+                                  SyntaxKind.RemoveAccessorDeclaration,
+                                  SyntaxKind.GetAccessorDeclaration,
+                                  SyntaxKind.SetAccessorDeclaration);
+
+        protected override Analyzer GetAnalyzer(CompilationStartAnalysisContext context, CompilationMSInternalTypes types)
+        {
+            Analyzer analyzer = new CSharpAnalyzer(types);
+            context.RegisterSyntaxNodeAction(analyzer.AnalyzeNode, nodeKindsOfInterest);
```

Comment:
As we discussed offline, I am not the author of this check, just moving the code now. I will create a bug for rewriting and cleaning up the rule.